### PR TITLE
Trip flare prototype

### PR DESCRIPTION
**Pull request by PabstMirror**
#394

How it works:
* Makes a new mine class that does almost no damage to player (but will still trigger an explode event)
* Adds an explosions EH on the defuse object we already have
* explsionEH spawns a flare when the mine blows up

It's all XEH driven, so it should handle disconnects/jip's fine.

Want to look at reducing camerashake from the "explosion".

### CODE REVIEW

**Comment by esteldunedain** [comment]
Brilliant

File: addons/explosives/functions/fnc_tripFlareOnPlace.sqf [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+EXPLODE_4_PVT(_this,_unit,_explosive,_magazineClass,_extra);
+
+//This should always be exactly one because we just created and attached to it.
+_attachedObjects = attachedObjects _explosive;
+
+if ((count _attachedObjects) != 1) exitWith {ERROR("attachedObjects count wrong");};
+
+_defuseHelper = _attachedObjects select 0;
+systemChat "Place";
```
Comment (KoffeinFlummi) [inline addons/explosives/functions/fnc_tripFlareOnPlace.sqf]: Debug

File: addons/explosives/functions/fnc_tripFlareExplosionEH.sqf [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+_explosive = _defuseHelper getVariable [QGVAR(Explosive), objNull];
+//Only explode if _explosive still exists and it took "lethal" damage
+//So we know the ExplosionEH was triggered by the mine exploding
+if ((isNull _explosive) || {(damage _explosive) < 1}) exitWith {};
+
+//Only do stuff it's a trip flare
+if (typeOf _explosive != "ACE_APERSTripFlare_Wire_Ammo") exitWith {};
+
+systemChat format ["Flare Triggered: [%1] from [%2]", _defuseHelper, _explosive];
```
Comment (KoffeinFlummi) [inline addons/explosives/functions/fnc_tripFlareExplosionEH.sqf]: Debu

**Comment by CorruptedHeart** [comment]
Do we need/want a separate model for this? Do we have someone who can do that?

**Comment by bux578** [comment]
>  Do we have someone who can do that?

iirc there's that guy on the public slack doing a model for @ulteq

**Comment by PabstMirror** [comment]
IMHO, the model isn't that bad (looks like a cylinder on a spike)

**Comment by ViperMaul** [comment]
Is this ready for testing and merge?

**Comment by PabstMirror** [comment]
This should be good to go.

**Comment by ulteq** [comment]
> Want to look at reducing camerashake from the "explosion".

I do not notice any cam shake from the tripwire explosion, but adding this to the ammo class should remove it completely:

`class CamShakeExplode {
    power = 0;
    duration = 0;
    frequency = 20;
    distance = 0;
};`

**Comment by ViperMaul** [comment]
I tested this and when placed it starts out flat, making you think it is bugged. Then when you arm it with a trigger then it sits upright. Is that by design? Any way to make it placed upright from the beginning?

**Comment by ViperMaul** [comment]
Any reason additional reasons why this was pushed to 3.2.0?

**Comment by bux578** [comment]
> I tested this and when placed it starts out flat, making you think it is bugged. Then when you arm it with a trigger then it sits upright. Is that by design? Any way to make it placed upright from the beginning?

The normal Tripwire Mine works the same way in ACE3.

**Comment by Mange** [comment]
I'd love to get this in. I could scrap my non-working prototype at [Mange/hehu_tripflare](https://github.com/Mange/hehu_tripflare).

I see that it isn't currently possible to merge this to `master`. Need any help to solve the conflicts?

**Comment by PabstMirror** [comment]
This relied on a dummy object that was attached to every explosive, they don't exist anymore in current version of ace; they are now created locally, on demand.

So this needs serious revision, or redoing.  I'm thinking of just creating an "dummy" object just on the server to watch for an explosive event.

**Comment by Mange** [comment]
Thanks for the heads-up!

**Pull request closed without merging**